core/unix/signal.c

```diff
@@ -5548,12 +5561,9 @@ execute_handler_from_cache(dcontext_t *dcontext, int sig, sigframe_rt_t *our_fra
     byte *xsp = get_sigstack_frame_ptr(dcontext, info, sig,
                                        our_frame/* take xsp from (translated)
                                                  * interruption point */);
-
-    sigcontext_t sc_interrupted = *sc;
-    dr_signal_action_t action = send_signal_to_client(
-        dcontext, sig, our_frame, sc_orig, access_address, false /*not blocked*/, f);
-    if (!handle_client_action_from_cache(dcontext, sig, action, our_frame, sc_orig,
-                                         &sc_interrupted, false /*!blocked*/))
+    if (!handle_sent_to_client_signal(dcontext, sig, our_frame, sc, sc_orig,
+                                      access_address, false /*not blocked*/, f,
+                                      false /*handle deliver*/))
         return false;
 
     LOG(THREAD, LOG_ASYNCH, 2, "execute_handler_from_cache for signal %d\n", sig);
```

Comment:
You said this PR shrinks the master_signal_handler_C space: but there's no change here to that function.  One of these must be inlined into it?

Reply:
There are code changes in master_signal_handler_C.

Reply:
Ah missed the function label in the diff on the first examination.  OK so that does reduce the usage on all record_ paths.